### ISSUE TITLE
Revise selling pitch

### ISSUE THREAD

**Issue by TWiStErRob** [burrunan/gradle-cache-action#72]
`gradle-command-action` -> `gradle-build-action` -> `gradle/actions/setup-gradle` evolution has happened over the years, and major versions were released, please revise if your points about improvements are still true.

Also please consider working with bigdaz to integrate them!

**Comment by rkrisztian** [comment]
I agree,  it's getting ever more confusing. Although with  v6, the caching functionality of `gradle/actions/setup-gradle` goes proprietary licensed—with worrisome terms and conditions—which makes me consider disabling the caching part and replacing with `burrunan/gradle-cache-action` instead:

https://blog.gradle.org/github-actions-for-gradle-v6

*Edit:* Actually, I'm also reading a section in the [`setup-gradle` documentation](https://github.com/gradle/actions/blob/main/docs/setup-gradle.md), that does not make the choice obvious to me:

> Prefer a 100% Open Source (MIT) path? We also provide a Basic Caching provider as a thin wrapper over actions/cache. This provider is free for all repositories (public and private) and can be enabled at any time by setting cache-provider: basic.

**Comment by rkrisztian** [comment]
Just to add to the confusion, `actions/setup-java@v5` also has basic Gradle caching support, although suggesting ` gradle/actions/setup-gradle` for advanced capabilities.  Personally, I think `setup-java` should focus on Java, leaving Gradle to `gradle/actions/setup-gradle`...

**Comment by vlsi** [comment]
Reviews on #159 are welcome
